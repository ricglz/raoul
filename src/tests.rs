use super::{parse, DirFunc};

#[test]
fn use_of_undeclared_variable() {
    let filename = "examples/invalid/undeclared-variable.ra";
    let program = std::fs::read_to_string(filename).expect(filename);
    let debug = true;
    let ast_response = parse(&program, debug);
    assert!(ast_response.is_ok());
    let ast = ast_response.unwrap();
    let mut dir_func = DirFunc::new();
    let build_res = dir_func.build_dir_func(ast);
    assert!(build_res.is_err());
}
