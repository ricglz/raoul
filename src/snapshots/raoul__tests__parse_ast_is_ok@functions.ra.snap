---
source: src/tests.rs
expression: quad_manager
input_file: src/examples/valid/functions.ra
---
0    - Goto       -     -     5
1    - Assignment 3000  -     1000
2    - EndProc    -     -     -
3    - Return     1000  -     -
4    - EndProc    -     -     -
5    - Era        1     1     -
6    - GoSub      1     -     -
7    - Era        1     3     -
8    - Param      3001  -     0
9    - GoSub      3     -     -
10   - Assignment 0     -     2000
11   - Assignment 2000  -     1000
12   - Print      1000  -     -
13   - PrintNl    -     -     -
14   - End        -     -     -

